### ISSUE TITLE
Explicitly document --features and their dependencies in Getting Started Guide

### ISSUE THREAD

**Issue by katrinabrock** [diesel-rs/diesel#1093]
I'm happy to submit a PR for this soon.

## Problem Description
"A Note on Installing diesel_cli" is helpful, but could be way more helpful to make install painless.

 Two main things
1. Possible arguments to `--features` (backends) should be explicitly listed. Maybe there is a cargo way to figure out what they are? Even if there is, it would be nice to explicitly list them in the getting started guide to make it rust-beginner friendly.

2. For each backend, it would be nice to list the exact package name needed for various OS package managers (`brew`, `apt-get`, `dnf`, do we need to cover Windows?)

## Checklist

- [X ] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.

**Comment by killercup** [comment]
Yes please! I'm not sure I'd put all that in the Getting Started guide, though. Maybe add a new document about _all_ the feature toggles Diesel has? The 3 backends are the relevant ones for the CLI, and a good start; but the same document could also talk about the features you need to enable in `diesel` (and `diesel-codegen`) to get DateTime, JSON, etc. support.

**Comment by katrinabrock** [comment]
For a guide to installing dependencies, do we need to cover windows or just Mac and common Linux flavors?

**Comment by killercup** [comment]
Eventually, we should cover every platform our users use. :) But we can absolutely start out with just Mac and Debian, for example!
